tests/cmake.py

```diff
@@ -8,10 +8,10 @@ def __init__(self, factory):
         self.runs = dict()
         self.factory = factory
 
-    def compile(self, targets, options=None):
+    def compile(self, targets, options={}):
```

Comment:
Shouldn't do this either.  Using `None` in the signature was correct.  Problem is that you now have a mutable object attached to the keyword argument, this is evaluated at import time and any modifications to this mutable object is seen my all calls to this function.
The classical, boilerplaty, pattern is to use `None` in the signature and then have:
```
if options is None:
    options = {}
```

Reply:
Also, totally minor nit, but since we have both set and dict literals I prefer to use the init functions for them when creating an empty one as that's much clearer: `options = dict()`